src/state/hass-base-mixin.ts

```diff
@@ -48,6 +48,12 @@ export class HassBaseEl extends LitElement {
       this._pendingHass = { ...this._pendingHass, ...obj };
       return;
     }
-    this.hass = { ...this.hass, ...obj };
+
+    // Object.assign and the spread operator only copy static values from a getter to the new object; Object.defineProperty must be used to retain the getter for hass.language
+    this.hass = Object.defineProperty({ ...this.hass, ...obj }, "language", {
+      get() {
+        return this.locale.language;
+      },
+    });
```

Comment:
We might want to reconsider using a getter, this is called a LOT, and I really don't want to make it any slower.

It would probably be better to update `language` whenever we update `locale`...

(also, we should not have the descriptor defined in 2 places, we should use `Object.getOwnPropertyDescriptors()`)